DevTools/HackStudio/Debugger/BacktraceModel.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2020, Itamar S. <itamar8910@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "BacktraceModel.h"
+#include "Debugger.h"
+
+RefPtr<BacktraceModel> BacktraceModel::create(const PtraceRegisters& regs)
+{
+    return adopt(*new BacktraceModel(create_backtrace(regs)));
+}
+
+GUI::Variant BacktraceModel::data(const GUI::ModelIndex& index, Role role) const
+{
+    if (role == Role::Display) {
+        auto& frame = m_frames.at(index.row());
+        return frame.function_name;
+    }
+    return {};
+}
+
+Vector<BacktraceModel::FrameInfo> BacktraceModel::create_backtrace(const PtraceRegisters& regs)
+{
+    u32 current_ebp = regs.ebp;
+    u32 current_instruction = regs.eip;
+    Vector<BacktraceModel::FrameInfo> frames;
+    do {
+        const auto& debug_info = Debugger::the().session()->debug_info();
```

Comment:
Perhaps it would be nicer to decouple this code from the globally available `DebugSession`.
We could pass the active `DebugSession&` as an argument to this function instead, and then this code doesn't need to make the assumption that `Debugger::the().session()` is non-null.